src/libsystemd/sd-daemon/sd-daemon.c

```diff
@@ -535,6 +534,12 @@ static int pid_notify_with_fds_internal(
         if (n_fds > 0 || send_ucred) {
                 struct cmsghdr *cmsg;
 
+                /* Let's make sure the multiplications below don't overflow, and also return a recognizable
+                 * error in case the caller tries to send more fds than the kernel limit. The kernel would
+                 * return EINVAL which is not too useful I'd say. */
+                if (n_fds > SCM_MAX_FD)
```

Comment:
Hmm, maybe do the check early using assert_return?

Reply:
assert_return() is really for programming errors of the users of our libraries. It will result in loging. I am not sure hitting this limit here is a programming error. i.e. of someone calls systemd-notify from the shell with a ridiculous number of fds, this should not be considered a programming error, hence I think assert_return( 9would be too much.

Reply:
At least the check can still be moved to the beginning of func? It's pointless to set up sockets and such if this restriction is hit?

Reply:
ok